src/components/interview.ts

```diff
@@ -36,47 +56,157 @@ export async function handleInterview(message: Discord.Message, args: string[],
+  let title = 'Available interviewers';
 
-  const outEmbed = new Discord.MessageEmbed().setColor('#0099ff').setTitle('Available Interviewers');
+  //checks if domain arg is passed and is valid, runs list without filters otherwise
+  if (!domain) {
+    res = await db.all('SELECT * FROM interviewers');
+  } else if (!(domain in available_domains)) {
+    await message.reply('Not a valid domain, valid domains are: ' + domainString());
+    return;
+  } else {
+    res = await db.all(
+      'SELECT * FROM interviewers WHERE user_id IN (SELECT user_id FROM domains WHERE domain = ?)',
+      domain
+    );
+    title = 'Available interviewers for ' + available_domains[domain];
+  }
+
+  //shuffles and prints the results thru embed
+  res = _.shuffle(res) as Interviewer[];
+  const outEmbed = new Discord.MessageEmbed().setColor('#0099ff').setTitle(title);
   let listString = '';
   for (let count = 0; count < RESULTS_PER_PAGE && count < res.length; count++) {
     const rows = res[count];
     listString +=
       '**' + (await client.users.fetch(rows['user_id'].toString())).tag + '** | [Calendar](' + rows['link'] + ')\n\n';
   }
   outEmbed.setDescription(listString);
-  await message.channel.send(outEmbed);
+  await message.reply(outEmbed);
+}
+
+async function clearProfile(message: Discord.Message): Promise<void> {
+  const db = await openDB();
+  const { id } = message.author;
+
+  //clear user data from both tables
+  await db.run('DELETE FROM interviewers WHERE user_id = ?', id);
+  await db.run('DELETE FROM domains WHERE user_id = ?', id);
+  await message.reply('Your interviewer data has been cleared!');
+}
+
+async function showProfile(message: Discord.Message): Promise<void> {
+  const db = await openDB();
+  const { id } = message.author;
+
+  //check if user signed up to be interviewer
+  const interviewer = await db.get('SELECT * FROM interviewers WHERE user_id = ?', id);
+  if (!interviewer) {
+    await message.reply("You don't seem to have signed up yet, run `.interviewer signup [link]`");
+    return;
+  }
+
+  //get user's domains (if any)
+  const res = await db.all('SELECT * FROM domains WHERE user_id = ?', id);
+  let domains = '';
+  let first_domain = true;
+  for (const i of res) {
+    if (!first_domain) {
+      domains += ', ';
+    }
+    first_domain = false;
+    domains += available_domains[i.domain];
+  }
+
+  //build output embed
+  const outEmbed = new Discord.MessageEmbed().setColor('#0099ff').setTitle('Interviewer Profile');
+  outEmbed.addField('**Link**', interviewer['link']);
+  outEmbed.addField('**Domains**', domains === '' ? 'None' : domains);
+  await message.reply(outEmbed);
+}
+
+async function addDomain(message: Discord.Message, args: string[]): Promise<void> {
+  const db = await openDB();
+  const { id } = message.author;
+
+  //check user signed up to be an interviewer
+  const interviewer = await db.get('SELECT * FROM interviewers WHERE user_id = ?', id);
```

Comment:
this check/reply is used in multiple places, maybe this could be its own helper?

Reply:
I'm not sure if that would make the code any simpler, since its already one line. I'll also have to overload it a ton since different sql commands have different variables (alternatively, if I did a format string, it would probably be more complex)

Reply:
If you're worried about the return type, you could just return a boolean and do the check inside the helper, some like `doesInterviewerExist`. Normally yes, this is already really simple, but generally it's good practice to not have the same SQL queries defined in multiple places. Does that make sense?